tardis/plasma/tests/test_nlte_solver.py

```diff
@@ -218,3 +226,88 @@ def test_jacobian_matrix(
         [0.0, 1.0, 0.0, 1.0, 2.0, -1.0],
     ]
     assert_almost_equal(actual_jacobian_matrix, desired_jacobian_matrix)
+
+
+@pytest.fixture(scope="session")
+def nlte_atomic_data_fname(tardis_ref_path):
+    """
+    File name for the atomic data file used in NTLE ionization solver tests.
+    """
+    atomic_data_fname = os.path.join(
+        tardis_ref_path, "nlte_atom_data", "TestNLTE_He_Ti.h5"
+    )
+
+    atom_data_missing_str = (
+        f"{atomic_data_fname} atomic datafiles " f"does not seem to exist"
+    )
+
+    if not os.path.exists(atomic_data_fname):
+        pytest.exit(atom_data_missing_str)
+
+    return atomic_data_fname
+
+
+@pytest.fixture(scope="session")
+def nlte_atomic_dataset(nlte_atomic_data_fname):
+    """
+    Atomic dataset used for NLTE ionization solver tests.
+    """
+    nlte_atomic_data = AtomData.from_hdf(nlte_atomic_data_fname)
+    return nlte_atomic_data
+
+
+@pytest.fixture
+def nlte_atom_data(nlte_atomic_dataset):
+
+    atomic_data = deepcopy(nlte_atomic_dataset)
+    return atomic_data
```

Comment:
Does anyone know why the deepcopy is needed? I know that it is also used for the normal atomic data but I don't understand why?

Reply:
not sure what happens with it afterwards. if it gets filtered we might loose information

Reply:
there is always one way of finding out what happens when we don't use `deepcopy` :-)

Reply:
The test seems to work just fine without the `deepcopy`, so I'll just remove this.
For the normal atomic data maybe because we use it in different places, that's why we need a deepcopy?

Reply:
Alright, so it turns out if you want to use the atomic dataset if different places for example, to assemble different plasmas, you need to use the deepcopy. Otherwise, the code throws an error that "Aromic Data was already prepared".